src/pysdmx/io/reader.py

```diff
@@ -0,0 +1,112 @@
+"""SDMX All formats reader module."""
+
+from enum import Enum
+from io import BytesIO
+from pathlib import Path
+from typing import Union
+
+from pysdmx.errors import Invalid
+from pysdmx.io.input_processor import process_string_to_read
+from pysdmx.model.dataset import Dataset
+from pysdmx.model.message import Message
+
+
+class ReadFormat(Enum):
+    """Enumeration of supported SDMX read formats."""
+
+    SDMX_ML_2_1 = "SDMX-ML 2.1"
+    # SDMX_JSON_2 = "SDMX-JSON 2.0.0"
```

Comment:
@sosna This is a placeholder to include as well emthods to read the JSON formats, still I have not found the way to implement them. Any ideas?